test/controllers/contactPreference/ChangeEmailControllerSpec.scala

```diff
@@ -0,0 +1,336 @@
+            }
+
+            "return HTML" in {
+              contentType(result)(defaultTimeout) shouldBe Some("text/html")
+              charset(result) shouldBe Some("utf-8")
+            }
+
+            "prepopulate the form with the validation email" in {
+              document.select("input").attr("value") shouldBe testValidationEmail
+            }
+            "not call the VatSubscription service" in {
+              verify(mockVatSubscriptionService, never())
+                .getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext])
+            }
+          }
+          "the previous form value is retrieved from session" should {
+
+            lazy val result = {
+              mockConfig.features.letterToConfirmedEmailEnabled(true)
+              target().show(request.withSession(
+                common.SessionKeys.validationEmailKey -> testValidationEmail,
+                common.SessionKeys.prepopulationEmailKey -> testValidEmail)
+              )
+            }
+            lazy val document = Jsoup.parse(bodyOf(result))
+
+            "return 200" in {
+              status(result) shouldBe Status.OK
+            }
+
+            "return HTML" in {
+              contentType(result) shouldBe Some("text/html")
+              charset(result) shouldBe Some("utf-8")
+            }
+
+            "prepopulate the form with the previously entered form value" in {
+              document.select("input").attr("value") shouldBe testValidEmail
+            }
+
+            "not call the VatSubscription service" in {
+              verify(mockVatSubscriptionService, never())
+                .getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext])
+            }
+          }
+        }
+
+        "there is no email in session" when {
+
+          "the customerInfo call succeeds" should {
+
+            lazy val result = target().show(request)
+            lazy val document = Jsoup.parse(bodyOf(result))
+
+            "return 200" in {
+              status(result) shouldBe Status.OK
+            }
+
+            "return HTML" in {
+              contentType(result) shouldBe Some("text/html")
+              charset(result) shouldBe Some("utf-8")
+            }
+
+            "prepopulate the form with the customerInfo result" in {
+              document.select("input").attr("value") shouldBe testValidEmail
+            }
+          }
+
+          "the customerInfo call fails" should {
+
+            lazy val result = target(Left(ErrorModel(
+              Status.NOT_FOUND,
+              "error"
+            ))).show(request)
+
+            "return 404" in {
+              status(result) shouldBe Status.INTERNAL_SERVER_ERROR
+            }
+
+            "return HTML" in {
+              contentType(result) shouldBe Some("text/html")
+              charset(result) shouldBe Some("utf-8")
+            }
+          }
+        }
+      }
+
+
+
+      "a user is does not have a valid enrolment" should {
```

Comment:
Grammar - "a user is does"

Reply:
I know this is tiny but it wasn't addressed the first time around, and while you're here again, would you mind?